bundler/lib/bundler/plugin.rb

```diff
@@ -47,6 +47,32 @@ def install(names, options)
       Bundler.ui.error "Failed to install plugin #{name}: #{e.message}\n  #{e.backtrace.join("\n ")}"
     end
 
+    # Uninstalls plugins by the given names
+    #
+    # @param [Array<String>] names the names of plugins to be uninstalled
+    def uninstall(names, options)
+      if names.empty? && !options[:all]
+        Bundler.ui.error "No plugins to uninstall. Specify at least 1 plugin to uninstall.\n"\
+          "Use --all option to uninstall all the installed plugins."
+        return
+      end
+
+      names = index.installed_plugins if options[:all]
+      if names.any?
+        names.each do |name|
+          if index.installed?(name)
+            Bundler.rm_rf(index.plugin_path(name))
+            index.unregister_plugin(name)
+            Bundler.ui.info "Uninstalled plugin #{name}"
+          else
+            Bundler.ui.error "Plugin #{name} is not installed \n"
+          end
+        end
+      else
+        Bundler.ui.info "No plugins installed"
+      end
```

Comment:
I wonder if we should simplify this method to do something symmetric to what the `install` method:

* Extract an `Plugin:Uninstaller` class which takes care of the actual file removal and returns the list of uninstalled plugins. Maybe this class could be reused for the `rescue` block logic of the `install` method?

* Move the unregistration of the uninstalled plugins returned by the `Uninstaller` to an `unsave_plugins` helper.

Do you think this refactoring is worth doing?

Reply:
Imo, this method is pretty simple right now and and we have better control with logging messages. I'd left it as it is for now.

Reply:
Ok, let's leave it for now.